gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-certificate/src/main/java/io/gravitee/am/plugins/certificate/core/impl/CertificatePluginManagerImpl.java

```diff
@@ -19,157 +19,59 @@
 import io.gravitee.am.certificate.api.CertificateConfiguration;
 import io.gravitee.am.certificate.api.CertificateMetadata;
 import io.gravitee.am.certificate.api.CertificateProvider;
-import io.gravitee.am.plugins.certificate.core.CertificateConfigurationFactory;
-import io.gravitee.am.plugins.certificate.core.CertificateDefinition;
 import io.gravitee.am.plugins.certificate.core.CertificatePluginManager;
-import io.gravitee.plugin.core.api.Plugin;
+import io.gravitee.am.plugins.certificate.core.CertificateProviderConfiguration;
+import io.gravitee.am.plugins.handlers.api.core.AMPluginManager;
+import io.gravitee.am.plugins.handlers.api.core.ConfigurationFactory;
 import io.gravitee.plugin.core.api.PluginContextFactory;
-import io.gravitee.plugin.core.internal.AnnotationBasedPluginContextConfigurer;
+import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.ConfigurableApplicationContext;
-import org.springframework.context.annotation.Import;
-
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.*;
 
 /**
  * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
  * @author GraviteeSource Team
  */
-public class CertificatePluginManagerImpl implements CertificatePluginManager {
+public class CertificatePluginManagerImpl extends CertificatePluginManager  {
```

Comment:
I think this would be better served as simply one class. There's only this one implementation and there's also the abstract`ProviderPluginManager` now.